src/ir/auth_logic/ast_construction_test.cc

```diff
@@ -0,0 +1,78 @@
+//-------------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//-------------------------------------------------------------------------------
+
+#include "src/ir/auth_logic/ast_construction.h"
+
+#include <iostream>
+#include <memory>
+#include <optional>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <variant>
+#include <vector>
+
+#include "src/common/testing/gtest.h"
+#include "src/ir/auth_logic/ast.h"
+#include "src/ir/datalog/program.h"
+
+namespace raksha::ir::auth_logic {
+
+TEST(AstConstructionSuite, SimpleQueryTest) {
+  absl::string_view input_text =
+      R"qql(may_notifierA = query "UserA" says mayA("NotifierA")
+                                         may_notifierB = query "UserB" says mayB("NotifierB"))qql";
+
+  Program prog_out = ParseProgram(input_text);
+  std::vector<std::string> actual_name;
+  std::vector<std::string> actual_principal_name;
+  std::vector<std::string> actual_predicate_name;
+  std::vector<std::string> actual_predicate_args;
```

Comment:
Don't you lose the correlation between different parts of the query by testing the components separately? I think you should directly test the list of query and use matchers (https://github.com/google/googletest/blob/main/docs/reference/matchers.md). It will make the tests more readable and easier to debug in case of failures. 

See suggestion below.